src/braiding/braiding.rs

```diff
@@ -36,8 +221,9 @@ impl Braid {
         }
 
         Braid {
-            state,
+            state: state.clone(),
```

Comment:
Need to re-look at if this is okay. Or if should implement copy or smthn for state.